apps/src/code-studio/components/header/SignInCalloutWrapper.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import cookies from 'js-cookie';
+import SignInCallout from './SignInCallout';
+
+const HideSignInCallout = 'hide_signin_callout';
+
+// The use of both session storage and cookies is to check for 1 day
+// and 1 session, and display the callout again only once BOTH have passed.
+export default class SignInCalloutWrapper extends React.Component {
+  constructor(props) {
+    super(props);
+    this.closeCallout = this.closeCallout.bind(this);
+    this.state = {
+      hideCallout:
+        cookies.get(HideSignInCallout) === 'true' ||
+        sessionStorage.getItem(HideSignInCallout) === 'true'
+    };
+  }
+
+  closeCallout(event) {
+    this.setState({hideCallout: true});
+    cookies.set(HideSignInCallout, 'true', {expires: 1, path: '/'});
```

Comment:
(very minor) I think '/' is the default path and can be omitted?

Reply:
I've done some StackOverflow research about this, and I think it does make a small difference for which directories are included in the cookies application. If you don't include one at all, then the cookie is only available in the exact directory it is recorded. Adding the '/' makes it available a directory higher, which I think we would want if we want to be able to access the cookie. I've also noticed it implemented this way in other places in our codebase (see apps/src/templates/SwipePrompt.jsx).

Reply:
Sounds good, thanks for doing the research and sharing your findings!